src/upload/index.ts

```diff
@@ -37,20 +49,24 @@ export const upload = (
         formData.append('file', uploadOptions.file, String(uploadOptions.fileName));
       } else if (key === "tags" && Array.isArray(uploadOptions.tags)) {
         formData.append('tags', uploadOptions.tags.join(","));
+        reqPayload.uploadPayload[key] = uploadOptions.tags.join(",");
       } else if (key === "responseFields" && Array.isArray(uploadOptions.responseFields)) {
         formData.append('responseFields', uploadOptions.responseFields.join(","));
+        reqPayload.uploadPayload[key] = uploadOptions.responseFields.join(",");
       } else if (key === "extensions" && Array.isArray(uploadOptions.extensions)) {
         formData.append('extensions', JSON.stringify(uploadOptions.extensions));
+        reqPayload.uploadPayload[key] = JSON.stringify(uploadOptions.extensions);
       } else if (key === "customMetadata" && typeof uploadOptions.customMetadata === "object" &&
         !Array.isArray(uploadOptions.customMetadata) && uploadOptions.customMetadata !== null) {
         formData.append('customMetadata', JSON.stringify(uploadOptions.customMetadata));
-      } else if(uploadOptions[key] !== undefined) {
+        reqPayload.uploadPayload[key] = JSON.stringify(uploadOptions.customMetadata);
+      } else if (uploadOptions[key] !== undefined) {
         formData.append(key, String(uploadOptions[key]));
+        if (key !== 'file')
+          reqPayload.uploadPayload[key] = String(uploadOptions[key]);
```

Comment:
Why is this needed? Isn't this case covered in the first if condition?

Reply:
when typeOf uploadOptions.file  is "string" then this will be required

Reply:
Resolved

Reply:
Resolved